Packs/AzureRiskyUsers/Integrations/AzureRiskyUsers/AzureRiskyUsers.py

```diff
@@ -223,7 +226,47 @@ def get_skip_token(next_link: str | None, outputs_prefix: str, outputs_key_field
         return parse_qs(parsed_url.query)['$skiptoken'][0]
 
 
-def risky_users_list_command(client: Client, args: dict[str, str]) -> CommandResults:
+def do_pagination(client: Client, response: dict[str, Any], limit: int = 1) -> dict:
+    """
+    Retrieves a limited number of pages by repeatedly making requests to the API using the nextLink URL
+    until it has reached the specified limit or there are no more pages to retrieve,
+    :param response: response body, contains collection of chat/message objects
+    :param limit: the requested limit
+    :return: dict of the limited response_data and the last nextLink URL.
+    """
+
+    response_data = response.get('value', [])
+    next_link = response.get('@odata.nextLink')
+    # old_next_link is for a situation that we have more results in the response then the limit
+    while (next_link := response.get('@odata.nextLink')) and len(response_data) < limit:
+        demisto.debug(f"Using response {next_link=}")
+        response = client.risky_users_list_request(limit=limit, skip_token=next_link)
+        response_data.extend(response.get('value', []))
+    demisto.debug(f'The limited response contains: {len(response_data[:limit])}')
+    return {'value': response_data[:limit], "@odata.context": response.get('@odata.context'), '@odata.nextLink': next_link}
```

Comment:
now we wouldn't miss any items of the next link because of the reason mentioned in the previous review?

Reply:
Yes, because we do not use the nextlink in the limit argument.